### ISSUE TITLE
Auto-failover mechnism for vineyard cluster on kubernetes

### ISSUE THREAD

**Issue by dashanji** [v6d-io/v6d#1175]
Describe your problem
---------------------

Although we have the CRD `Backup` to back up all objects in vineyard cluster and CRD `Recover` to recover them, users need to manage it manually. However, we can define some `Failover Strategy` in Vineyardd CRD as follows. 
```
EnableFailover: true
Backup:
    # Backup the vineyard cluster every 5 min
    period: 5min
```

The Vineyardd Controller will detect the status of vineyard cluster(Vineyardd) and trigger the failover operation by the two situation:
1. Node Failure.
2. Pod Restart.

In addition, when the `Recover` job finished, the old object id is deleted and a new object will be created. Uers can check the mapping relationship between old and new objects from the statue of `Recover` CR. In this case, it requires users to look up the mapping relationship every time to obtain the new object id. It will be much more convenient if usesr can obtain the data by reading the old object id.

In summary, the main tasks are as follows:
- [ ] Support regular failover in vineyardd controller.
- [ ] Support reading old and deleted object during `Failover`.

**Comment by sighingnow** [comment]
Closing as the cronjob is left for the workflow engine on top of vineyard cluster.
